pkg/provisioningv2/kubeconfig/manager.go

```diff
@@ -230,10 +249,53 @@ func (m *Manager) GetCRTBForClusterOwner(cluster *v1.Cluster, status v1.ClusterS
 	}, nil
 }
 
+// kubeConfigValid accepts a kubeconfig and corresponding data, and validates that the kubeconfig is valid for the
+// cluster in question. It returns two booleans, the first of which is whether an error occurred parsing the kubeconfig
+// or retrieving information related to the kubeconfig, and the second which indicates whether the kubeconfig is valid.
+func (m *Manager) kubeConfigValid(kcData []byte, cluster *v1.Cluster, currentServerURL, currentServerCA, currentManagementClusterName string) (bool, bool) {
+	if len(kcData) == 0 {
+		return true, false
+	}
+	kc, err := clientcmd.Load(kcData)
+	if err != nil {
+		logrus.Errorf("error while loading kubeconfig in kubeconfigmanager for validation: %v", err)
+		return true, false
+	}
+	var serverURL, managementCluster string
+	splitServer := strings.Split(kc.Clusters["cluster"].Server, "/k8s/clusters/")
+	if len(splitServer) != 2 {
+		return true, false
+	}
```

Comment:
```suggestion
	serverURL, managementCluster, found := strings.Cut(kc.Clusters["cluster"].Server, "/k8s/clusters/")
	if !found {
		return true, false
	}
```
Forever going to plug [strings.Cut](https://cs.opensource.google/go/go/+/go1.20.1:src/strings/strings.go;l=1262)